src/ts/controller/base.ts

```diff
@@ -0,0 +1,221 @@
+   */
+  idsOfDay: IDS_OF_DAY = {};
+
+  /****************
+   * CRUD Schedule
+   ****************/
+  /**
+   * Create an schedule instance from raw data.
+   * @param {ScheduleData} scheduleData - Data object to create schedule.
+   * @returns {Schedule[]} The instance of Schedule that created.
+   */
+  createSchedule(scheduleData: ScheduleData) {
+    const schedule = Schedule.create(scheduleData);
+
+    return this.addSchedule(schedule);
+  }
+
+  /**
+   * Add an schedule instance.
+   * @param {Schedule} schedule The instance of Schedule.
+   * @returns {Schedule} The instance of Schedule that added.
+   */
+  addSchedule(schedule: Schedule) {
+    this.schedules.add(schedule);
+    this._addToMatrix(schedule);
+
+    return schedule;
+  }
+
+  /**
+   * Create schedules from raw data.
+   * @param {ScheduleData[]} dataList - schedule data list to create schedule.
+   * @returns {Schedule[]} The instance list of Schedule that created.
+   */
+  createSchedules(schedules: ScheduleData[] = []) {
+    return schedules.map(scheduleData => {
+      return this.createSchedule(scheduleData);
+    });
+  }
+
+  /**
+   * Update an schedule.
+   */
+  updateSchedule(scheduleId: string, calendarId: string, scheduleData: ScheduleData) {
+    const schedule = this.schedules.single(item => {
+      return isSameSchedule(item, scheduleId, calendarId);
+    });
+
+    if (!schedule) {
+      return false;
+    }
+
+    schedule.init(scheduleData);
+
+    this._removeFromMatrix(schedule);
+    this._addToMatrix(schedule);
+
+    return true;
+  }
+
+  /**
+   * Delete schedule instance from controller.
+   * @param {Schedule} schedule - schedule instance to delete
+   * @returns {Schedule} deleted model instance.
+   */
+  deleteSchedule(schedule: Schedule) {
+    this._removeFromMatrix(schedule);
+    this.schedules.remove(schedule);
+
+    return schedule;
+  }
+
+  clearSchedules() {
+    this.idsOfDay = {};
+    this.schedules.clear();
+  }
+
+  /**
+   * Set calendar list
+   * @param {Array.<Calendar>} calendars - calendar list
+   */
+  setCalendars(calendars: CalendarData[]) {
+    this.calendars = calendars;
+  }
+
+  /**
+   * Return schedules in supplied date range.
+   *
+   * available only YMD.
+   * @param {TZDate} start start date.
+   * @param {TZDate} end end date.
+   * @returns {object.<string, Collection>} schedule collection grouped by dates.
+   */
+  findByDateRange(start: TZDate, end: TZDate): Record<string, Schedule[]> {
+    const range = getContainDatesInSchedule(start, end);
+    const ownSchedules = this.schedules.items;
+    const ownMatrix = this.idsOfDay;
+    const result: Record<string, Schedule[]> = {};
+    let ids;
+    let ymd;
+    let viewModels: Schedule[];
+
+    range.forEach(date => {
+      ymd = format(date, 'YYYYMMDD');
+      ids = ownMatrix[ymd];
+      viewModels = result[ymd] = [];
+
+      if (ids && ids.length) {
+        viewModels.push(...ids.map(id => ownSchedules[id]));
```

Comment:
```js
// before
viewModels.add.apply(viewModels, util.map(matrix, function(id) {
     return ScheduleViewModel.create(ownSchedules[id]);
}));
```
It's simple.
But, Don't need scheduleViewModel?

Reply:
yes, base controller's findByDateRange has not been used. So I remain it for future's using now. If it doesn't need for any view types, will be removed.